docs/pages/plugins_docs.md

```diff
@@ -0,0 +1,177 @@
+# MTT Plugin Overview
+To make the tool more modular and easier to generate tests, MTT makes use of the yapsy plugin system for python. Each stage that executes an .ini file makes use of at least one plugin. 
+
+Example:
+```
+[TestGet:HelloWorld]
+plugin = Copytree
+src = /opt/mtt/samples/python/
+```
+
+Plugins are divided into categories:
+-  Stages: Stages of test execution
+    - MTTDefaults
+    - BIOS
+    - Firmware
+    - Provision
+    - Profile
+    - MiddlewareGet
+    - MiddlewareBuild
+    - TestGet
+    - TestBuild
+    - LauncherDefaults
+    - TestRun
+    - Reporter
+-  System: Plugins essential for MTT to run
+-  Tools: Plugins required by Stages
+    - The Tools plugins are split into modules:
+        - Build
+        - CNC
+        - Executor
+        - Fetch
+        - Harraser
+        - Launcher
+        - Version
+-  Utilities: Plugins used by the MTT framework
+
+You can see the hierarchy where the plugins are defined along with their ordering on [Doxygen](/mtt/html/inherits.html). 
+
+### Stages
+The stages, as mentioned above, are used for stages. Some stages may contain additional plugins (i.e. [Reporter](https://github.com/open-mpi/mtt/tree/master/pylib/Stages/Reporter)). Each plugin in the Stages category has a set ordering which dictates what order the plugins should run.
+
+A list of stages and their ordering can be found on [Doxygen](/mtt/html/group__Stages.html)
+
+### System
+These plugins are essential for MTT to run. However, they are not actually plugins although we treat them as such. They simply extend the object class. For explanation purposes, we will continue to refer to them as plugins.
+- [LoadClasses.py](https://github.com/open-mpi/mtt/blob/master/pylib/System/LoadClasses.py)
+- [TestDef.py](https://github.com/open-mpi/mtt/blob/master/pylib/System/TestDef.py)
+
+LoadClasses is called by TestDef and loads all plugins by going through the directories and loading plugins to memory.
+
+TestDef is passed to all other plugins so that they have access to the log, to global plugins (i.e. logger, modcmd, execmd, etc. that are saved to TestDef so they are accessible by other plugins) and to the options. This plugin is important for several reasons:
+- It is the center point for setting configurations for MTT
+- Sets the configuration for tests
+- Logs results
+- Loads plugins by called LoadClasses
+- Starts execution of tests by called one of the [Executor](https://github.com/open-mpi/mtt/tree/master/pylib/Tools/Executor) plugins. The [combinatorial](https://github.com/open-mpi/mtt/blob/master/pylib/Tools/Executor/combinatorial.py) plugin is currently the default plugin; however, the [sequential](https://github.com/open-mpi/mtt/blob/master/pylib/Tools/Executor/sequential.py) plugin can be set by using the ```--executor=sequential``` flag.  
```

Comment:
The default executor is sequential:
https://github.com/open-mpi/mtt/blob/75cd103f305224ac995fad462057c1b34a5de9e6/pyclient/pymtt.py#L273-L277

Reply:
Ok! Fixed it. Thanks for pointing that out.